### PR TITLE
AB#2786 -- Updated phone number validation and added phone number parsing

### PR DESCRIPTION
**Pull request by AdamP-HRSDC-RHDCC**
Updated phone number validation to allow for more phone inputs:

Added parsing to convert all inputted phone numbers to the format of: "(111) 222-3333"

Supported phone number inputs:
1112223333
111 222 3333
111222 3333
111 2223333
111-222-3333
111 222-3333
111-222 3333
111222-3333
111-2223333
(111)2223333
(111) 222 3333
(111) 2223333
(111)222 3333
(111)-222-3333
(111) 222-3333
(111)-222 3333
(111)222-3333
(111)-2223333

### CODE REVIEW

**Comment by sebastien-comeau** [comment]
Wouldn't have been simpler to use [`libphonenumber-js`](https://github.com/catamphetamine/libphonenumber-js) based on Google's [libphonenumber](https://github.com/google/libphonenumber) and suggested by that Google repo (see bottom sectoin "Alternatives to our own version"). @gregory-j-baker, what do you think? We used it in one of YDG or CSC application before.

File: frontend/app/routes/_gcweb-app.update-phone-number.tsx [hunk 1]
```diff
@@ -14,8 +14,21 @@ export async function loader({ request }: LoaderFunctionArgs) {
   return json({ userInfo });
 }
 
+function parsePhoneNumber(phoneNumber: string): string {
```
Comment (Fbasham) [inline frontend/app/routes/_gcweb-app.update-phone-number.tsx:17]: Feel free to ignore, but this seems kind of clunky.  It should also go in a file under the `/utils` folder as well.
I feel a more 'all-encompassing' approach would be the following:

```javascript
function parsePhoneNumber(phoneNumber: string): string {
  let digits = phoneNumber.replace(/\D/g, '');
  return `(${digits.slice(0, 3)})${digits.slice(3, 6)}-${digits.slice(6)}`;
}
```

**Comment by gregory-j-baker** [comment]
> Wouldn't have been simpler to use [`libphonenumber-js`](https://github.com/catamphetamine/libphonenumber-js) based on Google's [libphonenumber](https://github.com/google/libphonenumber) and suggested by that Google repo (see bottom sectoin "Alternatives to our own version"). @gregory-j-baker, what do you think? We used it in one of YDG or CSC application before.

Libphonenumber is good, in my opinion..

**Comment by AdamP-HRSDC-RHDCC** [comment]
> > Wouldn't have been simpler to use [`libphonenumber-js`](https://github.com/catamphetamine/libphonenumber-js) based on Google's [libphonenumber](https://github.com/google/libphonenumber) and suggested by that Google repo (see bottom sectoin "Alternatives to our own version"). @gregory-j-baker, what do you think? We used it in one of YDG or CSC application before.
> 
> Libphonenumber is good, in my opinion..

I'll reimplement it with libphonenumber, also are we targeting only NANP or also international numbers?

File: frontend/app/routes/_gcweb-app.update-phone-number.tsx (hunk 1 above)
Comment (AdamP-HRSDC-RHDCC) [reply frontend/app/routes/_gcweb-app.update-phone-number.tsx]: Updated to use the libphonenumber library which uses formatNational() so a custom utils function is no longer needed.

**Comment by AdamP-HRSDC-RHDCC** [comment]
Updated to use the libphonenumber library

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**